app/authentication/auth.py

```diff
@@ -137,10 +137,19 @@ def requires_auth():
 
 
 def _auth_by_api_key(auth_token):
+    # TODO: uncomment this when the grace period for the token prefix is over
+    # orig_token = auth_token
+
     try:
         # take last 36 chars of string so that it works even if the full key is provided.
         auth_token = auth_token[-36:]
         api_key = get_api_key_by_secret(auth_token)
+
+        # TODO: uncomment this when the grace period for the token prefix is over
+        # check for token prefix
+        # if current_app.config["API_KEY_PREFIX"] not in orig_token:
+        #     raise AuthError("Invalid token: you must re-generate your API key to continue using GC Notify", 403, service_id=api_key.service.id, api_key_id=api_key.id)
```

Comment:
They will also get this error if they just send the secret part as we currently require. We should add something here to point this out.

Reply:
When you generate the key, admin gives you a single string that includes key name, svc id and secret id.  Are there docs somewhere that suggest to users not providing back exactly what admin spits out? 🤔

Reply:
yeah, right now we explicitly tell users to only put the last uuid in the header 😞 
https://documentation.notification.canada.ca/en/start.html#headers

Reply:
Very interesting.  While this does work when calling the API directly, it fails using any of the client libraries.  We should include this info in our communications when we are asking clients to regenerate their keys.  I'll make a note over it in that other card.